### PR TITLE
[AI Tutor] expose profanity and PII filter results to front end

### PR DESCRIPTION
**Pull request by Erin007**
Continuation of the work in #55656 
[CT-239](https://codedotorg.atlassian.net/browse/CT-239)

Here, I pass the results from the profanity filtering along to the frontend so that we can skip sending the inappropriate message to OpenAI, differentially style inappropriate messages (hide the content and tint red for profanity and yellow for PII) and provide a canned response from AI Tutor with specific feedback for the student about how to  make their message appropriate. 

https://github.com/code-dot-org/code-dot-org/assets/12300669/4b49eac5-1363-490f-8b93-80406ab012f5

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/aiTutor/redux/aiTutorRedux.ts [hunk 1]
```diff
@@ -77,9 +77,6 @@ export const submitChatMessage = createAsyncThunk(
     const systemPrompt = generalChatSystemPrompt;
     const storedMessages = state.aiTutor.chatMessages;
     const newMessageId = storedMessages[storedMessages.length - 1].id + 1;
-    const appropriateChatMessages = storedMessages.filter(
-      msg => msg.status === Status.OK
-    );
```
Comment (Erin007) [inline apps/src/aiTutor/redux/aiTutorRedux.ts:82]: We show all the messages now, just hide content for inappropriate messages and style them differently.

File: dashboard/app/controllers/openai_chat_controller.rb [hunk 2]
```diff
@@ -10,9 +10,11 @@ def chat_completion
 
     # Check for PII / Profanity
     locale = params[:locale] || "en"
-    messages = params[:messages].collect {|msg| msg[:content]}.join(' ')
-    filter_result = ShareFiltering.find_failure(messages, locale) if messages
-    return render(status: :bad_request, json: {type: filter_result.type, content: filter_result.content}) if filter_result
+    # Just look at the most recent message from the student.
+    message = params[:messages].last[:content]
```
Comment (molly-moen) [inline dashboard/app/controllers/openai_chat_controller.rb:14]: Why do we only need to look at the most recent message?

Comment (Erin007) [reply dashboard/app/controllers/openai_chat_controller.rb:14]: Any previous messages will have already been flagged and displayed as such in the UI. If we look for filter failures in the whole set of messages every time, then appropriate messages that come after inappropriate ones will be incorrectly flagged as inappropriate.

**Review by molly-moen — APPROVED** [review]
🎉

File: apps/src/aichat/types.ts [hunk 3]
```diff
@@ -26,12 +26,18 @@ export enum Role {
...
   ERROR = 'error',
+  PROFANITY = 'profanity',
+  PERSONAL = 'personal',
   INAPPROPRIATE = 'inappropriate',
   OK = 'ok',
-  PERSONAL = 'personal',
   UNKNOWN = 'unknown',
+  EMAIL = 'email',
+  ADDRESS = 'address',
+  PHONE = 'phone',
```
Comment (ebeastlake) [inline apps/src/aichat/types.ts:36]: super nit: "phone number"?

**Review by ebeastlake — APPROVED** [review]

File: apps/src/aichat/types.ts (hunk 3 above)
Comment (Erin007) [reply apps/src/aichat/types.ts:36]: I need these to match the [strings that the share filter returns](https://github.com/code-dot-org/code-dot-org/blob/cf3667238e3aeff62f1f3d8e10c6cac21a75f49a/lib/cdo/share_filtering.rb#L41).

**Pull request merged**